### PR TITLE
fix/Edit Administrations

### PR DESCRIPTION
**Pull request by ksmontville**
## Proposed changes
Implement watchers and invalidate queries in order to properly handle administration edits.

Firekit PR: https://github.com/yeatmanlab/roar-firekit/pull/140

## Types of changes

What types of changes does this pull request introduce?

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (non-breaking change that does not add functionality but makes code cleaner or more efficient)
- [ ] Documentation Update
- [ ] Tests (new or updated tests)
- [ ] Style (changes to code styling)
- [ ] CI (continuous integration changes)
- [ ] Repository Maintenance
- [ ] Other (please describe below)

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/yeatmanlab/roar-dashboard/blob/main/.github/CONTRIBUTING.md).
- [x] The changes in this PR are as small as they can be. They represent one and only one fix or enhancement.
- [x] Linting checks pass with my changes.
- [ ] Any existing unit tests pass with my changes.
- [x] Any existing end-to-end tests pass with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] If this PR fixes an existing issue, I have added a unit or end-to-end test that will detect if this issue reoccurs.
- [ ] I have added JSDoc comments as appropriate.
- [ ] I have added the necessary documentation to the [roar-docs repository](https://github.com/yeatmanlab/roar-docs).
- [ ] I have shared this PR on the roar-pr-reviews channel (if I have access)
- [ ] I have linked relevant issues (if any)

### CODE REVIEW

File: src/components/CreateAdministration.vue [hunk 1]
```diff
@@ -625,7 +677,6 @@ function findVariantWithParams(variants, params) {
     const cleanInputParams = removeNull(params);
     return _isEqual(cleanInputParams, cleanVariantParams);
   });
-
   if (found) {
     console.log('found', found);
```
Comment (Emily-ejag) [inline src/components/CreateAdministration.vue:681]:
```suggestion
 
```

File: src/components/CreateAdministration.vue [hunk 2]
```diff
@@ -625,7 +677,6 @@ function findVariantWithParams(variants, params) {
     const cleanInputParams = removeNull(params);
     return _isEqual(cleanInputParams, cleanVariantParams);
   });
-
   if (found) {
```
Comment (Emily-ejag) [inline src/components/CreateAdministration.vue:680]:
```suggestion

```

File: src/components/CreateAdministration.vue [hunk 3]
```diff
@@ -625,7 +677,6 @@ function findVariantWithParams(variants, params) {
...
     console.log('found', found);
   }
```
Comment (Emily-ejag) [inline src/components/CreateAdministration.vue:682]:
```suggestion

```

File: src/components/CreateAdministration.vue [hunk 4]
```diff
@@ -614,9 +622,53 @@ watch([preExistingAdminInfo, allVariants], ([adminInfo, allVariantInfo]) => {
...
+
+watch(familiesToGrab, async (updatedValue) => {
+  // Invalidate the families query and re-fetch the data based on the updated value
+  await queryClient.invalidateQueries(['families', props.adminId]);
+  preFamilies.value = queryClient.fetchQuery(['families', props.adminId], () => fetchDocsById(updatedValue));
+  state.families = (await preFamilies.value) ?? [];
+});
+
 function findVariantWithParams(variants, params) {
   console.log(`attempting to find variant of ${variants[0].task.id}`);
```
Comment (Emily-ejag) [inline src/components/CreateAdministration.vue:673]:
```suggestion

```

**Comment by Emily-ejag** [comment]
This is working so good! thanks Kyle. there is just a tiny thing that is not working and is when selecting `This administration does not require consent or assent forms` save and then try to edit, that particular selection is not showing, the rest is working fine.

**Review by Emily-ejag — CHANGES_REQUESTED** [review]
Also, sometimes on the first time I want to edit it does not show the `Selected organizations` and sometimes the names and dates. The warnings I see are the ones in the picture. But once I refresh It looks fine, and not warnings 
image

**Comment by ksmontville** [comment]
> This is working so good! thanks Kyle. there is just a tiny thing that is not working and is when selecting `This administration does not require consent or assent forms` save and then try to edit, that particular selection is not showing, the rest is working fine.

I fixed the checkbox issue and blank fields issue, do you mind taking another look?

**Comment by Emily-ejag** [comment]
It is awesome!! I think the only mark as test is not being pulled. But the rest is smooth like butter :) 
image

**Review by Emily-ejag — APPROVED** [review]

**Review by kellyel — APPROVED** [review]

**Review by richford — APPROVED** [review]

**Pull request merged**